src/graphics/render_stl.cpp

```diff
@@ -194,6 +198,7 @@ class Stl_context
 	void write_triangle(
 		const Triple& v1, const Triple& v2, const Triple& v3)
 	{
+        std::ostringstream ss;

```

Comment:
As mentioned, use same ostringstream for the whole object, or store facets in objects and convert to string late.